### PR TITLE
[new module] Azure CDN Profile module

### PR DESCRIPTION
**Pull request by caohai**
##### SUMMARY

- add azure_rm_cdnprofile module
- add azure_rm_cdnprofile_facts module
- add integration test for these two modules
- add cdn management client in azure_rm_common
- add necessary pkg info

##### ISSUE TYPE

 - New Module Pull Request
##### COMPONENT NAME

- azure_rm_cdnprofile
- azure_rm_cdnprofile_facts
##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (haicao-azure-cdnprofile 6b25649148) last updated 2018/08/16 01:59:39 (GMT +000)
  config file = None
  configured module search path = [u'/home/hai/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/hai/ansible/lib/ansible
  executable location = /home/hai/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+                    self.log("Need to update the CDN profile")
+
+                    if not self.check_mode:
+                        self.results = self.update_cdnprofile()
+                        self.log("Update done")
+
+                    self.results['changed'] = True
+                    return self.results
+
+        elif self.state == 'absent' and response:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py:202]: what  if self.state == absent and null response? any message to user?

File: lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        '''
+        Creates a Azure CDN profile.
+
+        :return: deserialized Azure CDN profile instance state dictionary
+        '''
+        self.log("Creating the Azure CDN profile instance {0}".format(self.name))
+
+        parameters = Profile(
+            location=self.location,
+            sku=Sku(name=self.sku),
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py:224]: self.sku none check

Comment (caohai) [reply lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py:224]: sku is required if state=='present'

File: lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py (hunk 1 above)
Comment (caohai) [reply lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py:202]: This means user is trying to delete an non-existed profile. No msg to user, but 'changed' is False in this situation. (If deletion is successful 'changed' is True)

**Comment by Fred-sun** [comment]
@caohai Thanks for the contribution, Could you help to confirm this PR ready for review or not? Thanks!

**Comment by Fred-sun** [comment]
@caohai Thanks for the contribution, Could you help to confirm this PR ready for review or not? Thanks!

**Comment by yungezz** [comment]
HI @mattclay ,  in this PR, we add CDN module with tests, while creating CDN profile failed with below error consistently. We tested manually in local and even in CI docker image:  drydock/u16pytall, cannot reproduce the error. could you pls help to check is there any environment related setting related to this failure? Thanks.
```
Error creating Azure CDN profile instance: Operation returned an invalid status code 'Conflict'
```

**Comment by mattclay** [comment]
@yungezz How are you running the tests? I'm seeing the same error when I run them with:

`ansible-test integration azure_rm_cdnprofile --docker -v`

**Comment by caohai** [comment]
Hi @mattclay, I run the test locally with:
```bash
ansible-test integration azure_rm_cdnprofile --docker -v
```
But no error msg occurred.

**Comment by yungezz** [comment]
@mattclay  as @caohai mentioned, we tried test with ansible-test, but no error met. only difference here is test account. Is there any special settings on the test account? meanwhile, we're trying to capture service request/response.

**Comment by Fred-sun** [comment]
@caohai Thanks for the contribution, Could you help confirm this PR ready for review or not? Thanks!

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/80024/74/tests

**Pull request closed without merging**